### PR TITLE
docs: adds container dependencies to docs

### PR DESCRIPTION
**Pull request by xar-tol**
Adds container dependencies to sidecar and job doc.

Adds container dependencies to docs to address #2175

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by huanjani — APPROVED** [review]
LGTM!

File: site/content/docs/developing/sidecars.en.md [hunk 1]
```diff
@@ -38,6 +38,9 @@ sidecars:
     # Optional Docker labels to apply to this container.
     labels:
       {label key} : <label value>
+    # Optional container dependencies to apply to this container.
+    depends_on:
+      {container name}: <status>
```
Comment (Lou1415926) [inline site/content/docs/developing/sidecars.en.md]: I wonder if 
```
      {container name}: <desired status>
```
or
```
      {container name}: <condition>
```
would help folks understand what the RHS want?

**Review by Lou1415926 — COMMENTED** [review]
Looks good! Sorry I may be missing something from previous PRs - should we add this to our doc for lb-web-service and backend-service as well?

File: site/content/docs/developing/sidecars.en.md (hunk 1 above)
Comment (xar-tol) [reply site/content/docs/developing/sidecars.en.md]: Good point! Adjusted phrasing and added the field to common-svc-fields to address your other note.

**Review by iamhopaul123 — APPROVED** [review]

**Review by Lou1415926 — APPROVED** [review]
:shipit:

**Pull request merged**